File: C++/Lanchonete/Cadastro/Proprietario.cpp
#include <iostream>
#include <vector>
#include "Proprietario.h"

using namespace std;

Proprietario::Proprietario(string &nome, string &email, string &senha) : Usuario(nome, email, senha) {
    this->lanchonetes = new vector<Lanchonete *>();
}

Proprietario::~Proprietario(){
    for(Lanchonete* l : *lanchonetes)
        delete l;
    delete lanchonetes;
}

Lanchonete *Proprietario::buscaLanchonetePorNome(string &nome) const {
    for (Lanchonete *l: *lanchonetes) {
        if (l->getNome() == nome)
            return l;
    }

    return nullptr;
}

void Proprietario::listarLanchonetes() const {
    cout << "\nLista de Lanchonetes" << endl;
    for (Lanchonete *l: *lanchonetes) {
        cout << *l << endl;
    }
}

int Proprietario::exibirMenu(DeliveryApp* app) {

    cout << "\n- Opções do Proprietário: -" << endl;
    cout << "1 - Cadastrar Lanchonete" << endl;
    cout << "2 - Remover Lanchonete" << endl;
    cout << "3 - Cadastrar Lanche em uma Lanchonete" << endl;
    cout << "4 - Visualizar Lista de Pedidos de uma Lanchonete" << endl;
    cout << "5 - Visualizar detalhes de um Pedido de uma Lanchonete" << endl;
    cout << "6 - Remover Pedido de uma Lanchonete" << endl;
    cout << "7 - Remover Cadastro" << endl;
    cout << "0 - Sair" << endl;
    cout << "Escolha uma opçao: ";

    int opcao;
    cin >> opcao;

    switch (opcao) {
        case 1:
            app->cadastrarLanchonete(this);
            break;
        case 2:
            app->removerLanchonete(this);
            break;
        case 3:
            app->cadastrarLanche(this);
            break;
        case 4:
            app->visualizarListaDePedidosLanchonete(this);
            break;
        case 5:
            app->visualizarDetalhesDeUmPedidoLanchonete(this);
            break;
        case 6:
            app->removerPedidoDeUmaLanchonete(this);
            break;
        case 7:
            app->removerCadastro(this);
            break;
        case 0:
            cout << "\nUsuário Desconectado!" << endl;
            break;
        default:
            cout << "\nOpção Inválida!" << endl;
            break;
    }

    return opcao;
}

bool Proprietario::verificarOpcaoValida(int opcao) {
    return opcao != 0 && opcao != 7;
}

string Proprietario::type(){
    return "Proprietario";
}

File: Java/Lanchonete/src/Cadastro/Proprietario.java
package Cadastro;

import Aplicativo.DeliveryApp;
import Aplicativo.Lanchonete;
import Excecoes.CodigoIdentificadorReplicadoException;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Scanner;

public class Proprietario extends Usuario {

    public ArrayList<Lanchonete> lanchonetes; //Lanchonetes que pertencem ao Proprietario

    public Proprietario(String nome, String email, String senha){
        super(nome, email, senha);
        lanchonetes = new ArrayList<>();
    }

    public Lanchonete buscaLanchonetePorNome(String nome){
        for(Lanchonete l : lanchonetes){
            if(l.getNome().equals(nome))
                return l;
        }

        return null;
    }

    //Lista todas as lanchonetes do Proprietario
    public void listarLanchonetes(){
        System.out.println("\nLista de Lanchonetes");
        for(Lanchonete l : lanchonetes){
            System.out.println(l);
        }
    }

    @Override
    public int exibirMenu(DeliveryApp app) throws IOException, CodigoIdentificadorReplicadoException {
        Scanner s = new Scanner(System.in);

        System.out.println("\n- Opções do Proprietário: -");
        System.out.println("1 - Cadastrar Lanchonete");
        System.out.println("2 - Remover Lanchonete");
        System.out.println("3 - Cadastrar Lanche em uma Lanchonete");
        System.out.println("4 - Visualizar Lista de Pedidos de uma Lanchonete");
        System.out.println("5 - Visualizar detalhes de um Pedido de uma Lanchonete");
        System.out.println("6 - Remover Pedido de uma Lanchonete");
        System.out.println("7 - Remover Cadastro");
        System.out.println("0 - Sair");
        System.out.print("Escolha uma opçao: ");

        int opcao = s.nextInt();

        switch(opcao){
            case 1 -> app.cadastrarLanchonete(this);
            case 2 -> app.removerLanchonete(this);
            case 3 -> app.cadastrarLanche(this);
            case 4 -> app.visualizarListaDePedidosLanchonete(this);
            case 5 -> app.visualizarDetalhesDeUmPedidoLanchonete(this);
            case 6 -> app.removerPedidoDeUmaLanchonete(this);
            case 7 -> app.removerCadastro(this);
            case 0 -> System.out.println("\nUsuário Desconectado!");
            default -> System.out.println("\nOpção Inválida!");
        }

        return opcao;
    }

    @Override
    public boolean verificarOpcaoValida(int opcao) {
        return opcao != 0 && opcao != 7;
    }

}
